manager/controlapi/service.go

```diff
@@ -239,6 +297,19 @@ func (s *Server) UpdateService(ctx context.Context, request *api.UpdateServiceRe
 	}
 
 	var service *api.Service
+	s.store.View(func(tx store.ReadTx) {
+		service = store.GetService(tx, request.ServiceID)
+	})
+	if service == nil {
+		return nil, grpc.Errorf(codes.NotFound, "service %s not found", request.ServiceID)
+	}
+
+	if request.Spec != nil && request.Spec.Endpoint != nil && (!reflect.DeepEqual(request.Spec.Endpoint, service.Endpoint) || !reflect.DeepEqual(request.Spec.Endpoint, service.Spec.Endpoint)) {
```

Comment:
Just compare `request.Spec.Endpoint` and `service.Spec.Endpoint` here. That will check if any changes are being made to the requested ports.

Also, it's not necessary to check if `request.Spec` is nil. `validateServiceSpec` already checks that.